pkg/controlplane/reconcilers/lease.go

```diff
@@ -307,4 +323,5 @@ func (r *leaseEndpointReconciler) StopReconciling() {
 	r.reconcilingLock.Lock()
 	defer r.reconcilingLock.Unlock()
 	r.stopReconcilingCalled = true
+	r.masterLeases.Destroy()
```

Comment:
some of the tests are dumping a lot of stack traces about grpc, can be related to this?
can we really destroy the storage?

Reply:
The storage is owned by Leases (and nothing else should be accessing it).

The `stopReconcilingCalled` when set to true, nothing more should try to access the underlying masterLeases, so conceptually it is safe to destroy it.

The test that are failing are failing with timeouts:
```
panic: test timed out after 10m0s
```

I have a feeling that it's something else, but let's hold the PR until we debug it.